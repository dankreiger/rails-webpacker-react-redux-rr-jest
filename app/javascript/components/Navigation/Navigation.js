import React, { Component } from 'react';
import { NavLink } from 'react-router-dom';
import {
  Collapse,
  Navbar,
  NavbarToggler,
  Nav,
  NavItem,
  UncontrolledDropdown,
  DropdownToggle,
  DropdownMenu,
  DropdownItem
} from 'reactstrap';

class Navigation extends Component {
  state = { isOpen: false };

  toggle = () => {
    this.setState({ isOpen: !this.state.isOpen });
  };
  render() {
    const { isOpen } = this.state;
    return (
      <Navbar color="light" light expand="md">
        <NavLink className="navbar-brand" to="/">
          Home
        </NavLink>
        <NavbarToggler onClick={this.toggle} />
        <Collapse isOpen={isOpen} navbar>
          <Nav className="ml-auto" navbar>
            <NavItem>
              <NavLink className="nav-link" to="/about">
                About
              </NavLink>
            </NavItem>
            <NavItem>
              <NavLink className="nav-link" to="/contact">
                Contact
              </NavLink>
            </NavItem>
          </Nav>
        </Collapse>
      </Navbar>
    );
  }
}

export default Navigation;
