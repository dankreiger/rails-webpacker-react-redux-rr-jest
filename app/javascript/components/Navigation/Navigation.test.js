import Navigation from './Navigation';
import { shallow } from 'enzyme';

describe('Navigation', () => {
  let navigationComponent;

  beforeEach(() => {
    navigationComponent = shallow(<Navigation />);
  });

  describe('rendering', () => {
    test('renders as expected', () => {
      expect(navigationComponent.length).toBe(1);
      expect(navigationComponent).toMatchSnapshot();
    });
  });
});
