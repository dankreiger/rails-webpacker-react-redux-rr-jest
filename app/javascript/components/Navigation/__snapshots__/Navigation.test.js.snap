// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation rendering renders as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Navigation />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <NavLink
          activeClassName="active"
          aria-current="page"
          className="navbar-brand"
          to="/"
        >
          Home
        </NavLink>,
        <NavbarToggler
          onClick={[Function]}
          tag="button"
          type="button"
        />,
        <Collapse
          appear={false}
          enter={true}
          exit={true}
          in={false}
          isOpen={false}
          mountOnEnter={false}
          navbar={true}
          onEnter={[Function]}
          onEntered={[Function]}
          onEntering={[Function]}
          onExit={[Function]}
          onExited={[Function]}
          onExiting={[Function]}
          tag="div"
          timeout={350}
          unmountOnExit={false}
        >
          <Nav
            className="ml-auto"
            navbar={true}
            tag="ul"
            vertical={false}
          >
            <NavItem
              tag="li"
            >
              <NavLink
                activeClassName="active"
                aria-current="page"
                className="nav-link"
                to="/about"
              >
                About
              </NavLink>
            </NavItem>
            <NavItem
              tag="li"
            >
              <NavLink
                activeClassName="active"
                aria-current="page"
                className="nav-link"
                to="/contact"
              >
                Contact
              </NavLink>
            </NavItem>
          </Nav>
        </Collapse>,
      ],
      "color": "light",
      "expand": "md",
      "light": true,
      "tag": "nav",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "aria-current": "page",
          "children": "Home",
          "className": "navbar-brand",
          "to": "/",
        },
        "ref": null,
        "rendered": "Home",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "onClick": [Function],
          "tag": "button",
          "type": "button",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "appear": false,
          "children": <Nav
            className="ml-auto"
            navbar={true}
            tag="ul"
            vertical={false}
          >
            <NavItem
              tag="li"
            >
              <NavLink
                activeClassName="active"
                aria-current="page"
                className="nav-link"
                to="/about"
              >
                About
              </NavLink>
            </NavItem>
            <NavItem
              tag="li"
            >
              <NavLink
                activeClassName="active"
                aria-current="page"
                className="nav-link"
                to="/contact"
              >
                Contact
              </NavLink>
            </NavItem>
          </Nav>,
          "enter": true,
          "exit": true,
          "in": false,
          "isOpen": false,
          "mountOnEnter": false,
          "navbar": true,
          "onEnter": [Function],
          "onEntered": [Function],
          "onEntering": [Function],
          "onExit": [Function],
          "onExited": [Function],
          "onExiting": [Function],
          "tag": "div",
          "timeout": 350,
          "unmountOnExit": false,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <NavItem
                tag="li"
              >
                <NavLink
                  activeClassName="active"
                  aria-current="page"
                  className="nav-link"
                  to="/about"
                >
                  About
                </NavLink>
              </NavItem>,
              <NavItem
                tag="li"
              >
                <NavLink
                  activeClassName="active"
                  aria-current="page"
                  className="nav-link"
                  to="/contact"
                >
                  Contact
                </NavLink>
              </NavItem>,
            ],
            "className": "ml-auto",
            "navbar": true,
            "tag": "ul",
            "vertical": false,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <NavLink
                  activeClassName="active"
                  aria-current="page"
                  className="nav-link"
                  to="/about"
                >
                  About
                </NavLink>,
                "tag": "li",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "active",
                  "aria-current": "page",
                  "children": "About",
                  "className": "nav-link",
                  "to": "/about",
                },
                "ref": null,
                "rendered": "About",
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <NavLink
                  activeClassName="active"
                  aria-current="page"
                  className="nav-link"
                  to="/contact"
                >
                  Contact
                </NavLink>,
                "tag": "li",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "active",
                  "aria-current": "page",
                  "children": "Contact",
                  "className": "nav-link",
                  "to": "/contact",
                },
                "ref": null,
                "rendered": "Contact",
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <NavLink
            activeClassName="active"
            aria-current="page"
            className="navbar-brand"
            to="/"
          >
            Home
          </NavLink>,
          <NavbarToggler
            onClick={[Function]}
            tag="button"
            type="button"
          />,
          <Collapse
            appear={false}
            enter={true}
            exit={true}
            in={false}
            isOpen={false}
            mountOnEnter={false}
            navbar={true}
            onEnter={[Function]}
            onEntered={[Function]}
            onEntering={[Function]}
            onExit={[Function]}
            onExited={[Function]}
            onExiting={[Function]}
            tag="div"
            timeout={350}
            unmountOnExit={false}
          >
            <Nav
              className="ml-auto"
              navbar={true}
              tag="ul"
              vertical={false}
            >
              <NavItem
                tag="li"
              >
                <NavLink
                  activeClassName="active"
                  aria-current="page"
                  className="nav-link"
                  to="/about"
                >
                  About
                </NavLink>
              </NavItem>
              <NavItem
                tag="li"
              >
                <NavLink
                  activeClassName="active"
                  aria-current="page"
                  className="nav-link"
                  to="/contact"
                >
                  Contact
                </NavLink>
              </NavItem>
            </Nav>
          </Collapse>,
        ],
        "color": "light",
        "expand": "md",
        "light": true,
        "tag": "nav",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "aria-current": "page",
            "children": "Home",
            "className": "navbar-brand",
            "to": "/",
          },
          "ref": null,
          "rendered": "Home",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "onClick": [Function],
            "tag": "button",
            "type": "button",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "appear": false,
            "children": <Nav
              className="ml-auto"
              navbar={true}
              tag="ul"
              vertical={false}
            >
              <NavItem
                tag="li"
              >
                <NavLink
                  activeClassName="active"
                  aria-current="page"
                  className="nav-link"
                  to="/about"
                >
                  About
                </NavLink>
              </NavItem>
              <NavItem
                tag="li"
              >
                <NavLink
                  activeClassName="active"
                  aria-current="page"
                  className="nav-link"
                  to="/contact"
                >
                  Contact
                </NavLink>
              </NavItem>
            </Nav>,
            "enter": true,
            "exit": true,
            "in": false,
            "isOpen": false,
            "mountOnEnter": false,
            "navbar": true,
            "onEnter": [Function],
            "onEntered": [Function],
            "onEntering": [Function],
            "onExit": [Function],
            "onExited": [Function],
            "onExiting": [Function],
            "tag": "div",
            "timeout": 350,
            "unmountOnExit": false,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <NavItem
                  tag="li"
                >
                  <NavLink
                    activeClassName="active"
                    aria-current="page"
                    className="nav-link"
                    to="/about"
                  >
                    About
                  </NavLink>
                </NavItem>,
                <NavItem
                  tag="li"
                >
                  <NavLink
                    activeClassName="active"
                    aria-current="page"
                    className="nav-link"
                    to="/contact"
                  >
                    Contact
                  </NavLink>
                </NavItem>,
              ],
              "className": "ml-auto",
              "navbar": true,
              "tag": "ul",
              "vertical": false,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": <NavLink
                    activeClassName="active"
                    aria-current="page"
                    className="nav-link"
                    to="/about"
                  >
                    About
                  </NavLink>,
                  "tag": "li",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "activeClassName": "active",
                    "aria-current": "page",
                    "children": "About",
                    "className": "nav-link",
                    "to": "/about",
                  },
                  "ref": null,
                  "rendered": "About",
                  "type": [Function],
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": <NavLink
                    activeClassName="active"
                    aria-current="page"
                    className="nav-link"
                    to="/contact"
                  >
                    Contact
                  </NavLink>,
                  "tag": "li",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "activeClassName": "active",
                    "aria-current": "page",
                    "children": "Contact",
                    "className": "nav-link",
                    "to": "/contact",
                  },
                  "ref": null,
                  "rendered": "Contact",
                  "type": [Function],
                },
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
